arangod/Aql/SubqueryStartExecutor.cpp

```diff
@@ -110,9 +110,9 @@ auto SubqueryStartExecutor::expectedNumberOfRows(size_t atMost) const
     // We will write one ShadowRow
     if (call.getLimit() > 0) {
       // We will write one DataRow
-      return 2;
+      return 2 * input.countDataRows();
     }
-    return 1;
+    return input.countDataRows();
```

Comment:
Aha, actually this is correct.
Because the inputRow will only be consumed if the executor has written it's shadow row.
=> we can only need to write a shadow row, if there is an input row left.